File: src/commonMain/kotlin/com/scoreo/ui/navigation/Screen.kt
package com.scoreo.ui.navigation

/**
 * Application screens (routes).
 * Pure data model, no Compose/DOM dependencies.
 */
sealed class Screen {
    data object Home : Screen()
    data object History : Screen()
    data object Import : Screen()
    data object Stats : Screen()
    data object Games : Screen()
    data object Sync : Screen()
    data class ScoreDetail(
        val gameTypeId: String,
        val playerIds: List<String>,
        val matchId: String? = null  // null = new match, non-null = edit existing
    ) : Screen()
}

File: src/ui/navigation/screen.ts
export type Screen =
  | { type: 'Home' }
  | { type: 'History' }
  | { type: 'Import' }
  | { type: 'Stats' }
  | { type: 'Games' }
  | { type: 'Sync' }
  | { type: 'ScoreDetail'; gameTypeId: string; playerIds: string[]; matchId?: string }

export const HOME_SCREEN: Screen = { type: 'Home' }
export const HISTORY_SCREEN: Screen = { type: 'History' }
export const IMPORT_SCREEN: Screen = { type: 'Import' }
export const STATS_SCREEN: Screen = { type: 'Stats' }
export const GAMES_SCREEN: Screen = { type: 'Games' }
export const SYNC_SCREEN: Screen = { type: 'Sync' }

export function scoreDetailScreen(gameTypeId: string, playerIds: string[], matchId?: string): Screen {
  return { type: 'ScoreDetail', gameTypeId, playerIds, matchId }
}
